### PR TITLE
Status dropdown: fix undefined values

### PR DESCRIPTION
**Pull request by AdrienClairembault**
When filtering the status dropdown, the values that didn't match the filter are shown are shown as "undefined" instead of being hidden from the results.

Before:

image

After:

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

File: js/common.js [hunk 1]
```diff
@@ -892,6 +892,11 @@ var templateSelection = function (selection) {
 var templateItilStatus = function(option) {
```
Comment (cedric-anne) [inline js/common.js:892]: First argument passed to callback of function referenced in `templateSelection` is `false` when element do not match selection. Not sure if it is a bug in select2 or not.
```suggestion
var templateItilStatus = function(option) {
    if (option === false) {
        // option is false when element does not match searched terms
        return null;
    }
```

File: js/common.js [hunk 2]
```diff
@@ -892,6 +892,11 @@ var templateSelection = function (selection) {
...
     var status = option.id || 0;
 
+    // Only show valid items
+    if (option.text == undefined) {
+        return null;
+    }
+
```
Comment (cedric-anne) [inline js/common.js:899]:
```suggestion
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]
All methods referenced in `templateSelection` option have to be fiexd. For instance, same bug appears when trying to filter priority.

**Comment by AdrienClairembault** [comment]
Looking into the others functions

**Comment by AdrienClairembault** [comment]
Done, I've also changed the priority rendering, I think the black dot wasn't displayed on purpose.

Before: 
image

After: 
image

This was it match the status dropdown that removes the icon for the "multi status" options.

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**